core/src/main/java/tc/oc/pgm/consumable/ConsumableMatchModule.java

```diff
@@ -0,0 +1,63 @@
+package tc.oc.pgm.consumable;
+
+import com.google.common.collect.ImmutableSet;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerItemConsumeEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.ParticipantState;
+import tc.oc.pgm.kits.tag.ItemTags;
+
+public class ConsumableMatchModule implements MatchModule, Listener {
+
+  private final Match match;
+  private final ImmutableSet<ConsumableDefinition> consumableDefinitions;
+
+  public ConsumableMatchModule(
+      Match match, ImmutableSet<ConsumableDefinition> consumableDefinitions) {
+    this.match = match;
+    this.consumableDefinitions = consumableDefinitions;
+  }
+
+  @EventHandler
+  private void onItemConsume(PlayerItemConsumeEvent event) {
+    ConsumableDefinition consumableDefinition = getConsumableDefinition(event.getItem());
+    if (consumableDefinition == null) return;
+    if (consumableDefinition.getCause() != ConsumeCause.EAT) return;
+
+    Player player = event.getPlayer();
+    MatchPlayer matchPlayer = match.getPlayer(player);
+    ParticipantState playerState = match.getParticipantState(player);
+    if (playerState == null) return;
```

Comment:
why do you want their participant state?

Reply:
I assumed this was so observers can't interact with consumables, which might be problematic for some actions.

Reply:
if you don't want observers to interact (and probably don't want dead people either), check if the player can interact with the match:
`ìf (!MatchPlayers.canInteract(matchPlayer)) return;`

participant state is not what you should use for that